packages/waku/src/lib/builder/output-cloudflare.ts

```diff
@@ -7,21 +7,18 @@ import { DIST_PUBLIC } from './constants.js';
 // XXX this can be very limited. FIXME if anyone has better knowledge.
 export const emitCloudflareOutput = async (
   rootDir: string,
-  config: ResolvedConfig,
-  serveJs: string,
+  config: ResolvedConfig
 ) => {
   const wranglerTomlFile = path.join(rootDir, 'wrangler.toml');
   if (!existsSync(wranglerTomlFile)) {
     writeFileSync(
       wranglerTomlFile,
       `
 name = "waku-project"
-main = "${config.distDir}/${serveJs}"
-compatibility_date = "2023-12-06"
-compatibility_flags = [ "nodejs_als" ]
+compatibility_date = "2024-04-05"
+compatibility_flags = ["nodejs_compat"]
 
-[site]
-bucket = "./${config.distDir}/${DIST_PUBLIC}"
+pages_build_output_dir = "./${config.distDir}/${DIST_PUBLIC}"
```

Comment:
Hm, I wonder if we can move public files.

```
pages_build_output_dir = "./output"
and
fs.rename("./${config.distDir}/${DIST_PUBLIC}", "./output")
```
something like that.

Reply:
Hmm. This one I'll probably have to leave to you to decide.

I think I had a similar idea. But I decided not to pursue it further, looked like it was getting very complex because the _worker.js fil then needs to be moved.

In other words, if we implement this strategy, then we must also move _worker.js to `/dist/public/output` as CFP requires _worker.js to exist in the root output directory. There's a good chance other import paths will also break after doing this?

Reply:
I'm not sure how much it would break others, but I would like to push it for now.
It eliminates the need of `waku.config.ts`, right?

We should revisit it anyway, when we design our build plugin system in the future (not sure when).

Reply:
In theory it should, I guess the only way is to try it. Let me do this soon and tell you the results :)

Yeah the way I see this solution, it's just to unblock people like me who want to start working immediately with Waku on a platform outside Vercel. I think even just having experimental support for CFP (& other platforms) will bring in more users instead of waiting for a full rewrite of the plugin system.

And users with existing Waku apps should be able to easily upgrade to the new adapter system when it is ready in future

Reply:
> Hm, I wonder if we can move public files.
> 
> ```
> pages_build_output_dir = "./output"
> and
> fs.rename("./${config.distDir}/${DIST_PUBLIC}", "./output")
> ```
> 
> something like that.

@dai-shi I need to clarify here, do you mean that `dist/public` should be renamed to `dist/output`? And then `pages_build_output_dir = "./dist/output"` Or am I misunderstanding

Reply:
I'm not sure what's the best, but what I'm suggesting is to restructure (rename) folders in `output-cloudflare.ts` and avoid changing `basePath`.

Reply:
Got it. Unfortunately the renaming from public to output doesn't look to work

![image](https://github.com/user-attachments/assets/17fe37ce-5a2d-4407-aedd-1ad4df243afc)

I also think it should be left as `public` as this is what [Vite expects](https://vitejs.dev/config/shared-options.html#publicdir)

@rmarscher any suggestions here? Did you encounter any challenges with the `public` directory?